io/src/main/scala/sbt/io/IO.scala

```diff
@@ -608,8 +608,22 @@ object IO {
    * If `file` or `base` are not absolute, they are first resolved against the current working directory.
    */
   def relativize(base: File, file: File): Option[String] = {
-    val basePath = (if (base.isAbsolute) base else base.getCanonicalFile).toPath.normalize()
-    val filePath = (if (file.isAbsolute) file else file.getCanonicalFile).toPath.normalize()
+    // "On UNIX systems, a pathname is absolute if its prefix is "/"."
+    // https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/File.html#isAbsolute
+    // "This typically involves removing redundant names such as "." and ".." from the pathname, resolving symbolic links (on UNIX platforms)"
+    // https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/File.html#getCanonicalPath()
+    // We do not want to use getCanonicalPath because if we resolve the symbolic link, that could change
+    // the outcome of copyDirectory's target structure.
```

Comment:
Not only that, symlink resolution is quite expensive (I haven't benchmarked getCanonicalFile, but Path.toRealPath is even slower than Files.readAttributes).